crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -172,30 +172,90 @@ impl Inner {
-            );
+            // automatic checks vs current implementation
+            match (settlement, auction_data) {
+                (Ok(_), None) => {
+                    // bug: we should have an auction_data
+                    tracing::warn!(?auction_id, "automatic check error: missing auction_data");
+                }
+                (Ok(settlement), Some(auction_data)) => {
+                    // staging settlement properly built
+                    let observation = settlement.observation();
+                    if observation.surplus.0 != auction_data.surplus {
+                        tracing::warn!(
+                            ?auction_id,
+                            ?observation.surplus,
+                            ?auction_data.surplus,
+                            "automatic check error: surplus mismatch"
+                        );
+                    }
+                    if observation.fee.0 != auction_data.fee {
+                        tracing::warn!(
+                            ?auction_id,
+                            ?observation.fee,
+                            ?auction_data.fee,
+                            "automatic check error: fee mismatch"
+                        );
+                    }
+                    if observation.order_fees.len() != auction_data.order_executions.len() {
+                        tracing::warn!(
+                            ?auction_id,
+                            ?observation.order_fees,
+                            ?auction_data.order_executions,
+                            "automatic check error: order_fees mismatch"
+                        );
+                    }
+                    for fee in auction_data.order_executions {
+                        if !observation
+                            .order_fees
+                            .contains_key(&domain::OrderUid(fee.0 .0))
+                        {
+                            tracing::warn!(
+                                ?auction_id,
+                                ?fee,
+                                ?observation.order_fees,
+                                "automatic check error: order_fees missing"
+                            );
+                        } else {
+                            let observation_fee =
+                                observation.order_fees[&domain::OrderUid(fee.0 .0)];
+                            if observation_fee.unwrap_or_default().0 != fee.1 {
+                                tracing::warn!(
+                                    ?auction_id,
+                                    ?observation_fee,
+                                    ?fee,
+                                    "automatic check error: order_fees value mismatch"
+                                );
+                            }
+                        }
+                    }
+                }
+                (Err(_), None) => {
+                    // todo check auction_ids are equal
```

Comment:
Waits for https://github.com/cowprotocol/services/pull/2866 to be merged so i can fetch auction_id from Settlement error